### ISSUE TITLE
Incorrect Implementations of Schematic interface

### ISSUE THREAD

**Issue by elhadiahmed** [angular/devkit#75]
I get the below lint error in the SchematicImpl class in [schematic.ts](https://github.com/angular/devkit/blob/master/packages/angular_devkit/schematics/src/engine/schematic.ts)
```Class 'SchematicImpl<CollectionT, SchematicT>' incorrectly implements interface 'Schematic<CollectionT, SchematicT>'.
  Types of property 'call' are incompatible.
    Type '<OptionT extends object>(options: OptionT, host: Observable<Tree>) => Observable<Tree>' is not assignable to type '<T>(options: T, host: Observable<Tree>) => Observable<Tree>'.
      Types of parameters 'options' and 'options' are incompatible.
        Type 'T' is not assignable to type 'object'.```

**Comment by hansl** [comment]
The right syntax for automatically closing issues in github is `Fixes #123`.

This issue is fixed by https://github.com/angular/devkit/pull/76

**Comment by elhadiahmed** [comment]
Noted.
